### PR TITLE
DEM-1875 overlay component

### PR DESCRIPTION
**Pull request by AngelGenchev95**
Dem-1875

### CODE REVIEW

File: src/assets/scss/components.scss [hunk 1]
```diff
@@ -263,3 +263,26 @@
     left: -1 * $circle-size - 8px;
   }
 }
+
+.p-overlay-container {
+  position:relative;
```
Comment (RusiPapazov) [inline src/assets/scss/components.scss:268]:
```suggestion
  position: relative;
```

**Review by T3ch0o — APPROVED** [review]

File: src/stories/Overlay/Overlay.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import figmaDecorator from 'storybook-addon-figma';
+import { storiesOf } from '@storybook/react';
+import { jsxDecorator } from 'storybook-addon-jsx';
+import { Overlay, Text } from '@paper/components';
+
+const buildOverlay = () => (
+  <Overlay content={<Text>Overlay text</Text>} open={true}>
```
Comment (RusiPapazov) [inline src/stories/Overlay/Overlay.stories.tsx:8]:
```suggestion
  <Overlay content={<Text>Overlay text</Text>}>
```

File: src/components/Overlay/enums/OverlayPositioningEnum.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export enum OVERLAY_POSITIONING {
+  CENTERED = 'centered',
```
Comment (RusiPapazov) [inline src/components/Overlay/enums/OverlayPositioningEnum.ts:2]: Want to add this here as well - do we need a positioning enum if it's only one and will possibly never change to two options?

Comment (AngelGenchev95) [reply src/components/Overlay/enums/OverlayPositioningEnum.ts:2]: I feel that we might have some more positioning in the future so it makes sense to have it preplanned. I guess @T3ch0o  and @EZarkov can also chime in here

Comment (RusiPapazov) [reply src/components/Overlay/enums/OverlayPositioningEnum.ts:2]: You what they say - YAGNI 😄 . But I'm ok with it either way.

Comment (EZarkov) [reply src/components/Overlay/enums/OverlayPositioningEnum.ts:2]: @AngelGenchev95 Here @RusiPapazov Is 100% right We will implement the positioning we needed. KISS & YAGNI. There is a better way to achieve it. The content will be div with 100% width and 100% height and inside it, the inner content can be placed how it is needed. Please remove this option for now.

**Review by EZarkov — CHANGES_REQUESTED** [review]

**Review by EZarkov — APPROVED** [review]

**Pull request merged**